pkg/collector/corechecks/cluster/helm/helm.go

```diff
@@ -0,0 +1,148 @@
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2021 Datadog, Inc.
+
+//go:build kubeapiserver
+// +build kubeapiserver
+
+package helm
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/cluster"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	checkName               = "helm"
+	maximumWaitForAPIServer = 10 * time.Second
+	informerSyncTimeout     = 60 * time.Second
+)
+
+func init() {
+	core.RegisterCheck(checkName, factory)
+}
+
+// HelmCheck collects information about the Helm releases deployed in the
+// cluster. The check only works for Helm installations configured to use
+// Kubernetes secrets as the storage. K8s secrets are the default in Helm v3.
+// Helm v2 used config maps by default. Ref:
+// https://helm.sh/docs/faq/changes_since_helm2/#secrets-as-the-default-storage-driver
+type HelmCheck struct {
+	core.CheckBase
+	secretLister      v1.SecretLister
+	runLeaderElection bool
+}
+
+var helmSecretsSelector = labels.Set{"owner": "helm"}.AsSelector()
+
+func factory() check.Check {
+	return &HelmCheck{
+		CheckBase:         core.NewCheckBase(checkName),
+		runLeaderElection: !config.IsCLCRunner(),
+	}
+}
+
+// Configure configures the Helm check
+func (hc *HelmCheck) Configure(config, initConfig integration.Data, source string) error {
+	apiCtx, apiCancel := context.WithTimeout(context.Background(), maximumWaitForAPIServer)
+	defer apiCancel()
+
+	apiClient, err := apiserver.WaitForAPIClient(apiCtx)
+	if err != nil {
+		return err
+	}
+
+	stopCh := make(chan struct{})
+	apiClient.InformerFactory.Start(stopCh)
+
+	informer := apiClient.InformerFactory.Core().V1().Secrets()
+	hc.secretLister = informer.Lister()
+	go informer.Informer().Run(stopCh)
+
+	return apiserver.SyncInformers(
+		map[apiserver.InformerName]cache.SharedInformer{"helm": informer.Informer()},
+		informerSyncTimeout,
+	)
+}
+
+// Run executes the check
+func (hc *HelmCheck) Run() error {
+	sender, err := aggregator.GetSender(hc.ID())
+	if err != nil {
+		return err
+	}
+	defer sender.Commit()
+
+	if hc.runLeaderElection {
+		isCurrentLeader, errLeader := isLeader()
+		if errLeader != nil {
+			return errLeader
+		}
+
+		if !isCurrentLeader {
+			log.Debugf("Not leader. Skipping the Helm check")
+			return nil
+		}
+	}
+
+	secrets, err := hc.secretLister.List(helmSecretsSelector)
```

Comment:
Note that this gets all the secrets owned by Helm.

This means that it also reports the superseeded releases (the ones that are no longer active after an upgrade) with the tag `helm_status=superseeded` . I'm not sure if we should limit this someone or just leave this limit to the Helm configuration.

Reply:
IMO we should keep all the release even if superseeded